functional/FunctionalTest.java

```diff
@@ -2280,89 +2274,84 @@ public static void presignedGetObject_test3() throws Exception {
       mintSuccessLog(
-          "presignedGetObject(PresignedGetObjectArgs args",
-          "expiry : 3600 sec, reqParams : response-content-type as application/json",
+          methodName,
+          "presigned get object expiry : 3600 sec, reqParams : response-content-type as application/json",
           startTime);
     } catch (Exception e) {
-      mintFailedLog(
-          "presignedGetObject(PresignedGetObjectArgs args)",
-          "expiry : 3600 sec, reqParams : response-content-type as application/json",
+      handleException(
+          methodName,
+          "presigned get object expiry : 3600 sec, reqParams : response-content-type as application/json",
           startTime,
-          null,
-          e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
-      throw e;
+          e);
     }
   }
 
-  /** public String presignedPutObject(PresignedPutObjectArgs). */
-  public static void presignedPutObject_test1() throws Exception {
+  /** public String getPresignedObjectUrl(GetPresignedObjectUrlArgs args). */
+  public static void getPresignedObjectUrl_test4() throws Exception {
+    String methodName = "getPresignedObjectUrl(GetPresignedObjectUrlArgs args)";
     if (!mintEnv) {
-      System.out.println("Test: presignedPutObject(PresignedPutObjectArgs args)");
+      System.out.println("Test: presigned put object: " + methodName);
     }
 
     long startTime = System.currentTimeMillis();
     try {
       String objectName = getRandomName();
       String urlString =
-          client.presignedPutObject(
-              PresignedPutObjectArgs.builder().bucket(bucketName).object(objectName).build());
+          client.getPresignedObjectUrl(
+              GetPresignedObjectUrlArgs.builder()
+                  .method(Method.PUT)
+                  .bucket(bucketName)
+                  .object(objectName)
+                  .build());
       byte[] data = "hello, world".getBytes(StandardCharsets.UTF_8);
       writeObject(urlString, data);
       client.removeObject(RemoveObjectArgs.builder().bucket(bucketName).object(objectName).build());
-      mintSuccessLog("presignedPutObject(String bucketName, String objectName)", null, startTime);
+      mintSuccessLog(methodName, "presigned put object", startTime);
     } catch (Exception e) {
-      mintFailedLog(
-          "presignedPutObject(PresignedPutObjectArgs args)",
-          null,
-          startTime,
-          null,
-          e.toString() + " >>> " + Arrays.toString(e.getStackTrace()));
-      throw e;
+      handleException(methodName, "presigned put object", startTime, e);
     }
   }
 
-  /** Test: presignedPutObject(PresignedPutObjectArgs args)). */
-  public static void presignedPutObject_test2() throws Exception {
+  /** Test: getPresignedObjectUrl(GetPresignedObjectUrlArgs args). */
+  public static void getPresignedObjectUrl_test5() throws Exception {
+    String methodName = "getPresignedObjectUrl(GetPresignedObjectUrlArgs args)";
```

Comment:
Please add an `args` variable similar to the previous suggestions and use it in the logging / exception-handling method calls.